### PR TITLE
Distinguish ambiguous column value of `None`

### LINKED ISSUE #479: Suggestion: distinguish ambiguous column value of `None`
We found there are three cases of getting `None` for column value.

1. When the user set the column value as `"NULL"`.
https://github.com/julien-duponchelle/python-mysql-replication/blob/8e17fb2f198676939a247f71275631db9834d4d6/pymysqlreplication/row_event.py#L158-L159
2. When the datatype is related to date(`DATETIME`, `TIME`, ...) and the value's format is not supportable.
https://github.com/julien-duponchelle/python-mysql-replication/blob/8e17fb2f198676939a247f71275631db9834d4d6/pymysqlreplication/row_event.py#L382-L394
3. When the datatype is `SET` and the value is empty
https://github.com/julien-duponchelle/python-mysql-replication/blob/8e17fb2f198676939a247f71275631db9834d4d6/pymysqlreplication/row_event.py#L246-L257

Since these values have the same value in python-mysql-replication as `None`, users cannot distinguish these cases.
Is it fine to distinguish these cases when `dump()` is executed like below?

```SQL
CREATE TABLE test_table (col0 int, col1 varchar(10), col2 datetime);
INSERT INTO test_table VALUES (1, 'abc', '2023-09-09 00:00:00');
UPDATE test_table SET col1=NULL, col2='0000-00-00 00:00:00' WHERE col0=1;
```
```python
# dump result for UPDATE

=== UpdateRowsEvent ===
Date: 2023-09-09T05:23:14
Log position: 1360
Event size: 37
Read bytes: 13
Table: test.test_table
Affected columns: 3
Changed rows: 1
Affected columns: 3
Values:
--
*col0:1=>1
*col1:abc=>None(null)
*col2:2023-09-09 00:00:00=>None(out of datetime range)
```

Here's a draft of our work : https://github.com/23-OSSCA-python-mysql-replication/python-mysql-replication/pull/86
I'd appreciate for your feedback :)

### PR DESCRIPTION
**Pull request by heehehe**
Resolve #479 

## Overview
In RowsEvent, there are three cases in which the column value is `None`.

1. when the value is `NULL`
2. when the datatype is related to time (datetime, time, ...) and the value is not supported format
(ex: '0000-00-00 00:00:00')
3. when the datatype is set and the value is empty

Since we cannot distinguish these cases, we're going to provide detail information for `None` value.

## Changes
- In RowsEvent, we added `__none_sources` variable.
- When we execute `__read_values_name`,
we check detail information for `None` cases and save it to `__none_sources`.
- When `dump()` executes, we check for none_sources and show them together with column name and value.

## Example

- SQL query
    ```sql
    CREATE TABLE test_table (col0 int, col1 varchar(10), col2 datetime);
    INSERT INTO test_table VALUES (1, 'abc', '2023-09-09 00:00:00');
    UPDATE test_table SET col1=NULL, col2='0000-00-00 00:00:00' WHERE col0=1;
    ```
- dump result for UPDATE
    ```diff
      === UpdateRowsEvent ===
      Date: 2023-09-09T05:23:14
      Log position: 1360
      Event size: 37
      Read bytes: 13
      Table: test.test_table
      Affected columns: 3
      Changed rows: 1
      Affected columns: 3
      Values:
      --
      *col0:1=>1
    + *col1:abc=>None(null)
    + *col2:2023-09-09 00:00:00=>None(out of datetime range)
    ```

### CODE REVIEW

File: pymysqlreplication/row_event.py [hunk 1]
```diff
@@ -177,17 +183,26 @@ def __read_values_name(
         elif column.type == FIELD_TYPE.BLOB:
             return self.__read_string(column.length_size, column)
         elif column.type == FIELD_TYPE.DATETIME:
-            return self.__read_datetime()
+            ret = self.__read_datetime()
+            if ret is None:
+                self.__none_sources[name] = "out of datetime range"
```
[the remark this answers is not in the document]
Comment (heehehe) [reply pymysqlreplication/row_event.py:188]: @julien-duponchelle Thanks for your review!

Do you mean that setting constant like
OUTDATETIME = "out of datetime range" 

and set __none_source like below?
self.__none_source[name] = OUTDATETIME

If this is right, I'm going to set constant by enum and set values with that :)

[the remark this answers is not in the document]
Comment (julien-duponchelle) [reply pymysqlreplication/row_event.py:188]: Yes

[the remark this answers is not in the document]
Comment (heehehe) [reply pymysqlreplication/row_event.py]: I added `pymysqlreplication/constants/NONE_SOURCE.py` 
and changed `row_event.py` to handle values according to the variables in `NONE_SORUCE.py`.

Please let me know if there's anything need to change :)

[the remark this answers is not in the document]
Comment (heehehe) [reply pymysqlreplication/row_event.py]: @julien-duponchelle 
I added `pymysqlreplication/constants/NONE_SOURCE.py` 
and changed `row_event.py` to handle values according to the variables in `NONE_SORUCE.py`.

Please let me know if there's anything need to change :)

**Review by sean-k1 — APPROVED** [review]
LGTM!
Thanks for hard work!

**Pull request merged**